### ISSUE TITLE
BIT fails to start on Arch

### ISSUE THREAD

**Issue by murygin** [bit-team/backintime#965]
BIT no longer starts on my currently patched Arch Linux. I installed version [1.1.24-1 from the AUR](https://aur.archlinux.org/packages/backintime/). 

```
[dm@sirius ~]$ backintime

Back In Time
Version: 1.1.24

Back In Time comes with ABSOLUTELY NO WARRANTY.
This is free software, and you are welcome to redistribute it
under certain conditions; type `backintime --license' for details.

Traceback (most recent call last):
  File "/usr/lib/python3.7/site-packages/dbus/bus.py", line 175, in activate_name_owner
    return self.get_name_owner(bus_name)
  File "/usr/lib/python3.7/site-packages/dbus/bus.py", line 361, in get_name_owner
    's', (bus_name,), **keywords)
  File "/usr/lib/python3.7/site-packages/dbus/connection.py", line 651, in call_blocking
    message, timeout)
dbus.exceptions.DBusException: org.freedesktop.DBus.Error.NameHasNoOwner: Could not get owner of name 'net.launchpad.backintime.serviceHelper': no such name

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/usr/share/backintime/common/backintime.py", line 874, in <module>
    start_app()
  File "/usr/share/backintime/common/backintime.py", line 463, in start_app
    return getConfig(args, False)
  File "/usr/share/backintime/common/backintime.py", line 584, in getConfig
    cfg = config.Config(args.config)
  File "/usr/share/backintime/common/config.py", line 272, in __init__
    self.setupUdev = tools.SetupUdev()
  File "/usr/share/backintime/common/tools.py", line 1213, in __init__
    conn = bus.get_object(SetupUdev.CONNECTION, SetupUdev.OBJECT)
  File "/usr/lib/python3.7/site-packages/dbus/bus.py", line 241, in get_object
    follow_name_owner_changes=follow_name_owner_changes)
  File "/usr/lib/python3.7/site-packages/dbus/proxies.py", line 248, in __init__
    self._named_service = conn.activate_name_owner(bus_name)
  File "/usr/lib/python3.7/site-packages/dbus/bus.py", line 180, in activate_name_owner
    self.start_service_by_name(bus_name)
  File "/usr/lib/python3.7/site-packages/dbus/bus.py", line 278, in start_service_by_name
    'su', (bus_name, flags)))
  File "/usr/lib/python3.7/site-packages/dbus/connection.py", line 651, in call_blocking
    message, timeout)
dbus.exceptions.DBusException: org.freedesktop.DBus.Error.Spawn.ChildExited: Launch helper exited with unknown return code 1
```

Is there a work around for this error?

**Comment by graysky2** [comment]
What version of python-qt4 are you using?  https://github.com/bit-team/backintime/issues/964

**Comment by murygin** [comment]
Probably the same problem as in issue #964.

**Comment by deesnook** [comment]
Same issue on Arch-based Manjaro 18.0

**Comment by sylph1o** [comment]
I have the same issue (on Arch Linux) and tried to build from the Git repository. The `make test` failed two tests with the following output.

```
python3 -m unittest  -b test/test_backintime.py
FF
Stderr:
/home/sylphio/tmp/backintime/common/test/test_backintime.py:33: DeprecationWarning: Please use assertEqual instead.
  self.assertEquals("", subprocess.getoutput("python3 backintime.py --quiet"))

======================================================================
FAIL: test_local_snapshot_is_successful (test.test_backintime.TestBackInTime)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/sylphio/tmp/backintime/common/test/test_backintime.py", line 68, in test_local_snapshot_is_successful
    self.assertEqual(proc.returncode, 0, msg)
AssertionError: 1 != 0 : Returncode: 1
stderr: Traceback (most recent call last):
  File "/usr/lib/python3.7/site-packages/dbus/bus.py", line 175, in activate_name_owner
    return self.get_name_owner(bus_name)
  File "/usr/lib/python3.7/site-packages/dbus/bus.py", line 361, in get_name_owner
    's', (bus_name,), **keywords)
  File "/usr/lib/python3.7/site-packages/dbus/connection.py", line 651, in call_blocking
    message, timeout)
dbus.exceptions.DBusException: org.freedesktop.DBus.Error.NameHasNoOwner: Could not get owner of name 'net.launchpad.backintime.serviceHelper': no such name

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/home/sylphio/tmp/backintime/common/backintime.py", line 1165, in <module>
    startApp()
  File "/home/sylphio/tmp/backintime/common/backintime.py", line 517, in startApp
    args.func(args)
  File "/home/sylphio/tmp/backintime/common/backintime.py", line 1150, in checkConfig
    cfg = getConfig(args)
  File "/home/sylphio/tmp/backintime/common/backintime.py", line 669, in getConfig
    cfg = config.Config(config_path = args.config, data_path = args.share_path)
  File "/home/sylphio/tmp/backintime/common/config.py", line 279, in __init__
    self.setupUdev = tools.SetupUdev()
  File "/home/sylphio/tmp/backintime/common/tools.py", line 1699, in __init__
    conn = bus.get_object(SetupUdev.CONNECTION, SetupUdev.OBJECT)
  File "/usr/lib/python3.7/site-packages/dbus/bus.py", line 241, in get_object
    follow_name_owner_changes=follow_name_owner_changes)
  File "/usr/lib/python3.7/site-packages/dbus/proxies.py", line 248, in __init__
    self._named_service = conn.activate_name_owner(bus_name)
  File "/usr/lib/python3.7/site-packages/dbus/bus.py", line 180, in activate_name_owner
    self.start_service_by_name(bus_name)
  File "/usr/lib/python3.7/site-packages/dbus/bus.py", line 278, in start_service_by_name
    'su', (bus_name, flags)))
  File "/usr/lib/python3.7/site-packages/dbus/connection.py", line 651, in call_blocking
    message, timeout)
dbus.exceptions.DBusException: org.freedesktop.DBus.Error.Spawn.ChildExited: Launch helper exited with unknown return code 1

stdout: 
Back In Time
Version: 1.2.0~alpha0 git branch 'master' hash 'ba37b1e'

Back In Time comes with ABSOLUTELY NO WARRANTY.
This is free software, and you are welcome to redistribute it
under certain conditions; type `backintime --license' for details.

INFO: Update to config version 6


======================================================================
FAIL: test_quiet_mode (test.test_backintime.TestBackInTime)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/sylphio/tmp/backintime/common/test/test_backintime.py", line 33, in test_quiet_mode
    self.assertEquals("", subprocess.getoutput("python3 backintime.py --quiet"))
AssertionError: '' != 'Traceback (most recent call last):\n  Fil[1939 chars]de 1'
Diff is 2013 characters long. Set self.maxDiff to None to see it.

Stderr:
/home/sylphio/tmp/backintime/common/test/test_backintime.py:33: DeprecationWarning: Please use assertEqual instead.
  self.assertEquals("", subprocess.getoutput("python3 backintime.py --quiet"))

----------------------------------------------------------------------
```

**Comment by alp82** [comment]
Did anyone find a solution?

**Comment by deesnook** [comment]
As a workaround, I switched to the Qt5 version. In the Manjaro repositories it's v1.1.12.r449.gba37b1e7-2. Not the same version, but it works

**Comment by thardeck** [comment]
Is not the main issue of this bug that backintime should use qt5 instead of the long unsupported qt4? If not I will open a new github issue for this topic.

openSUSE Tumbleweed will also soon remove all qt4 packages and I suppose this is the case for all other distributions too.

**Comment by deesnook** [comment]
I guess so. There's no current version of BIT using qt5 in the Arch repositories right now, just the version I mentioned above

**Comment by graysky2** [comment]
I am happily running [backintime-git](https://aur.archlinux.org/packages/backintime-git/) which has none of the unsupported qt4 depends/pure qt5.  Try it.  Hopefully, @Germar will create a pure QT5 official release soon!

**Comment by thardeck** [comment]
I have also tried the current master tarball build against openSUSE Tumbleweed and it works fine without qt4.
As you said, only the release is missing.

**Comment by Germar** [comment]
I finally released 1.2.0 today :wink:

**Comment by graysky2** [comment]
> I finally released 1.2.0 today wink

Got the announcement, thank you!  AUR pkg just [updated](https://aur.archlinux.org/cgit/aur.git/commit/?h=backintime&id=54850d0293ee18d0db1a0b2c4ec08d2297aa8acc).
